devicetypes/smartthings/zwave-siren.src/zwave-siren.groovy

```diff
@@ -84,23 +84,41 @@ metadata {
 	}
 }
 
+// Perform a periodic check to ensure that initialization of the device was successful
+def getInitVerifyCheckPeriodicSecs() {
+	30
+}
+
+def getInitVerifyCheckMaxAttempts() {
+	3
+}
+
 def installed() {
 	log.debug "installed()"
 	// Device-Watch simply pings if no device events received for 122min(checkInterval)
 	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, isStateChanged: true, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	state.initPeriodCheckAttempts = 0;
 	initialize()
 }
 
 def updated() {
 	log.debug "updated()"
 	state.configured = false
+	state.initPeriodCheckAttempts = 0;
 	// Device-Watch simply pings if no device events received for 122min(checkInterval)
 	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, isStateChanged: true, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
-	runIn(12, "initialize", [overwrite: true, forceForLocallyExecuting: true])
```

Comment:
Why did you remove the `runIn`? Part of the reason it's there is because `updated` can be called multiple times and this ensures the commands in `initialize` are only sent once (due to `overwrite: true`)

Reply:
The previous implementation would trigger an initialization step in 12 seconds. I removed the 12 second and call `initialize()` right away. This in my opinion is cleaner since the delay in the first place was needed to perform initialization and also there will not be a timer created if there is no need for initialization. 

In `initialize()`, the `runIn()` will still perform `overwrite: True` when creating the timer. `runIn(initVerifyCheckPeriodicSecs, "initialize", [overwrite: true, forceForLocallyExecuting: true])`

Perhaps, your question is that if `updated` is called while `runIn()` timer is pending, in that case, we sort of do double initialization. I guess I can revert it to `runIn(12, "initialize", [overwrite: true, forceForLocallyExecuting: true])` at `updated()` for that case. 

If there is